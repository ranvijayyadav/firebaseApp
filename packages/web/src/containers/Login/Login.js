import React, { Component } from 'react';
//import { withRouter } from 'react-router-dom';
import { firebase } from './../../firebase';
import { UserConsumer } from '../../context/UserContext'
import './Login.css';

class Login extends Component {
	state = {
		email: '',
		password: '',
		error: null,
	};
	handleInputChange = (event) => {
		this.setState({ [event.target.name]: event.target.value });
	};
	handleSubmit = (event, login) => {
		event.preventDefault();
		const { email, password } = this.state;
		firebase
			.auth()
			.signInWithEmailAndPassword(email, password)
			.then((user) => {
				login(user)
				this.props.history.push('/');
			})
			.catch((error) => {
				this.setState({ error: error });
			});
	};
	render() {
		if (this.state.error) {
			throw new Error(this.state.error);
		}
		const { email, password } = this.state;
		return (

			<UserConsumer>{({ login }) =>

				<div className="auth-wrapper">
					<div className="auth-inner">

						<form className="login-form" onSubmit={(event) => this.handleSubmit(event, login)}>

							<h3>Log in</h3>

							<div className="form-group">
								<label>Email</label>
								<input type="email" className="form-control" placeholder="Enter email" name="email" value={email} onChange={this.handleInputChange} />
							</div>

							<div className="form-group">
								<label>Password</label>
								<input type="password" name="password" value={password} onChange={this.handleInputChange} className="form-control" placeholder="Enter password" />
							</div>

							<button type="submit" className="btn btn-dark btn-lg btn-block log-in">Log in</button>
						</form>

					</div>
				</div>
			}
			</UserConsumer>
		);
	}
}
export default Login;//withRouter(Login);