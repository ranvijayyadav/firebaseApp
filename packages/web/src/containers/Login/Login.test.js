import { shallow, mount } from 'enzyme'
import Login from './Login';

describe('Login', () => {
    it('It renders properly', () => {
        const wrapper = shallow(<Login />)
        expect(wrapper).toMatchSnapshot();
    });
})