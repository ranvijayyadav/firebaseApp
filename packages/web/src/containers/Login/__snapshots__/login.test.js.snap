// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login It renders properly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
